src/data-management/data-model/RDF/rdf.ts

```diff
@@ -0,0 +1,266 @@
+import { DataFactory, Prefixes, Quad, Store } from "n3";
+import { Fetch } from "../../../fetch";
+import {
+  createContainer,
+  deleteSPARQLUpdate,
+  insertSPARQLUpdate,
+  patchSPARQLUpdate,
+  readParseResource,
+} from "../../Utils/modify-pod";
+import { TYPE_A } from "../namespace";
+
+const { namedNode, literal } = DataFactory;
+
+/**
+ * Represents an RDF resource.
+ */
+export class Rdf {
+  protected dataset: Store = new Store();
+  protected prefixes: Prefixes = {};
+
+  constructor(
+    readonly uri: string,
+    protected fetch: Fetch,
+    dataset?: Store,
+    prefixes?: Prefixes,
+  ) {
+    if (dataset) this.dataset = dataset;
+    if (prefixes) this.prefixes = prefixes;
+  }
+
+  /**
+   * Retrieves the type of the subject.
+   * @returns An array of strings representing the type of the subject, or undefined if not found.
+   */
+  getTypeOfSubject(): string[] | undefined {
+    return this.getObjectValuesFromPredicate(TYPE_A);
+  }
+
+  /**
+   * Retrieves the value of an object from a given predicate.
+   * @param predicate The predicate to retrieve the value from.
+   * @returns The value of the object, or undefined if no value is found.
+   */
+  getObjectValueFromPredicate(predicate: string): string | undefined {
+    const values = this.getObjectValuesFromPredicate(predicate);
+    if (values && values.length == 1) {
+      return values[0];
+    }
+    return undefined;
+  }
+
+  /**
+   * Retrieves the values of the objects that match the given predicate.
+   *
+   * @param predicate - The predicate to match against.
+   * @returns An array of string values if there are matching objects, otherwise undefined.
+   */
+  getObjectValuesFromPredicate(predicate: string): string[] | undefined {
+    const quads = this.dataset.match(namedNode(this.uri), namedNode(predicate));
+    const values: string[] = [];
+    for (const quad of quads) {
+      values.push(quad.object.value);
+    }
+
+    return values.length != 0 ? values : undefined;
+  }
```

Comment:
Det giver mest mening at retunere en tom liste, hvis listen er tom
```suggestion
  getObjectValuesFromPredicate(predicate: string): string[] {
    const quads = this.dataset.match(namedNode(this.uri), namedNode(predicate));
    const values: string[] = [];
    for (const quad of quads) {
      values.push(quad.object.value);
    }

    return values;
  }
```